app/src/main/java/org/mozilla/focus/browser/HomeTileRecyclerView.kt

```diff
@@ -97,9 +97,20 @@ class HomeTileAdapter(
         return null
     }
 
-    fun addItemToAdapter(homeTile: HomeTile) {
-        tiles.add(homeTile)
-        notifyItemInserted(tiles.lastIndex)
+    fun updateAdapterSingleInsertion(homeTiles: MutableList<HomeTile>) {
+        var i = 0
```

Comment:
Hm, it looks like you're doing a lot of checks here rather than thinking about the problem, and this could be a little more efficient. (If this were an interview question, I'd 100% ask for this code to be more concise :P)

- why not use a `for` loop instead of `while`? that handles the incrementing for you
- how about returning if you find and notify the item inserted, rather than continuing to iterate through even after you've found the item?
- why do you need the last check on size?